.ci/release/Jenkinsfile

```diff
@@ -309,3 +343,11 @@ def notifyStatus(def args = [:]) {
                       subject: args.subject,
                       body: args.body)
 }
+
+
+def cacheStage(def args = [:], Closure body){
+  // credentials are tied to the internal-ci controller that should have access to the beats-ci-temp bucket
+  stageStatusCache(id: args.id, bucket: 'beats-ci-temp', credentialsId: 'internal-ci-gcs-plugin-file-credentials', runAlways: params.force_build){
```

Comment:
Rather than persisting the status and running resuming from the build from there, could we just add a build parameter that allows to run from a specific stage? For example `start_from=release-wait-artifact-available-central` would skip the first few stages and resume from release-wait-artifact-available-central onwards.

This seems to be less complex and more predictable rather than persisting and reading from an object store.

Reply:
I just put another PR regarding an interactive approach:
- https://github.com/elastic/apm-agent-java/pull/2789. (still in progress, as it requires an extra input for gathering some parameters, and verify it will work as expected locally)

Rather than rerunning from a particular stage, it will ask for confirmation for each stage if needed.

Reply:
I also raised another PR with a different approach, for now I'll close this one and gather details in the private GitHub issue